src/main/java/com/codegym/shopyy/controller/api/CategoryController.java

```diff
@@ -0,0 +1,116 @@
+package com.codegym.shopyy.controller.api;
+
+import com.codegym.shopyy.dto.request.CategoryRequestDto;
+import com.codegym.shopyy.dto.response.ResponsePage;
+import com.codegym.shopyy.model.Category;
+import com.codegym.shopyy.model.SubCategory;
+import com.codegym.shopyy.service.impl.CategoryServiceImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/category")
+@CrossOrigin(origins = "*")
+public class CategoryController {
+
+    @Autowired
+    private CategoryServiceImpl categoryService;
+
+    @GetMapping
+    public ResponseEntity<Page<Category>> homeProduct(@RequestParam(defaultValue = "", required = false) String search,
+                                                      @PageableDefault(page = 0, size = 3) Pageable pageable) {
+
+        Page<Category> categoryPage;
+        if (!search.isEmpty()) {

```

Comment:
change !search.isEmpty() to search.isPresent()

Reply:
complete